src/components/event.js

```diff
@@ -60,23 +64,28 @@ class Event {
         this.chatEngine.pubnub.publish({
             message: m,
             channel: this.channel
-        }, (status) => {
+        }, (status, response) => {
 
             if (status.statusCode === 200) {
 
+                if (response) {
+                    m.timetoken = response.timetoken;
+                }
+
                 /**
                  * Message successfully published
                  * @event Chat#$"."publish"."success
```

Comment:
Another comment nitpick, wouldn't the ```@event Chat#$"."publish"."success``` look more like ```@event Chat#$."emitted```?

Reply:
yep